packages/zowe-explorer-api/src/fs/BaseProvider.ts

```diff
@@ -479,7 +479,7 @@ export class BaseProvider {
         return entry;
     }
 
-    protected _lookupParentDirectory(uri: vscode.Uri, silent?: boolean): DirEntry {
+    public _lookupParentDirectory(uri: vscode.Uri, silent?: boolean): DirEntry {
```

Comment:
Super nitpick, but now this function is public, the name should not have the underscore before it.
```suggestion
    public lookupParentDirectory(uri: vscode.Uri, silent?: boolean): DirEntry {
```

Reply:
couldn't we do something like this?

```typescript
    /**
     * @deprecated Please use the public version
     */
    protected _lookupParentDirectory(uri: vscode.Uri, silent?: boolean): DirEntry {
        return this.lookupParentDirectory(uri, silent);
    }
    public lookupParentDirectory(uri: vscode.Uri, silent?: boolean): DirEntry {
        return this._lookupAsDirectory(
            uri.with({
                path: path.posix.join(uri.path, ".."),
            }),
            silent ?? false
        );
    }
```

Reply:
in either scenario, we might want to adda changelog entry for the ze-api package. 😋

Reply:
We should start thinking on cleaning up some things before V4 😉 
I vote for adding this in the (future) list of breaking changes in the V4 changelog. 😋 

All that to say, yeah... We should replace all pre-existing references to use the new function 🙏